src/site/markdown/usage.md.vm

```diff
@@ -0,0 +1,259 @@
+title: Usage
+autor: Stephen Connolly
+date: 2008-09-02
+
+<!---
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## Velocity uses # as a prefix for macros and ## as a prefix for comments
+#set($h1 = '#')
+#set($h2 = '##')
+#set($h3 = '###')
+#set($h4 = '####')
```

Comment:
not used

Reply:
Yes, it's being used, see e.g.

https://github.com/ajarmoniuk/versions-maven-plugin/blob/273bb8ba8f3f6ce88126df4f501a12d247e89221/src/site/markdown/usage.md.vm#L28

https://github.com/ajarmoniuk/versions-maven-plugin/blob/273bb8ba8f3f6ce88126df4f501a12d247e89221/src/site/markdown/usage.md.vm#L33

https://github.com/ajarmoniuk/versions-maven-plugin/blob/273bb8ba8f3f6ce88126df4f501a12d247e89221/src/site/markdown/usage.md.vm#L69

etc :)

Reply:
I see, only `h4` is not used 😄

Reply:
Removed h4 :)

It's a pity there's h3 though, otherwise I could have used ---- and ===.

Reply:
Another idea for it https://issues.apache.org/jira/browse/DOXIA-677

Reply:
Regarding Doxia.

While working on those report renderers, I thought that it would actually be much better to be able to parse a velocity macro with some context injected, and simply render that context. Now we're doing that from the renderer class injecting some primitives into the sink, which "prints" everything using velocity, but much of velocity's potential goes to waste this way.